packages/tanstack-start/src/client/useAwaitableNavigate.ts

```diff
@@ -0,0 +1,31 @@
+import type { NavigateOptions } from '@tanstack/react-router';
+import { useLocation, useNavigate } from '@tanstack/react-router';
+import React from 'react';
+
+type Resolve = (value?: unknown) => void;
+
+export const useAwaitableNavigate = () => {
+  const navigate = useNavigate();
+  const location = useLocation();
+  const resolveFunctionsRef = React.useRef<Resolve[]>([]);
```

Comment:
❓@octoper I remember explictly not using refs for awaitable navigate for nextjs. Is this case different ?

Reply:
I followed what we already have in the Remix SDK as it's similar